src/lib/models/ConfigModel.ts

```diff
@@ -10,48 +11,85 @@ export interface ConfigModel {
 
 export type ConfigModelGeneratorFunction = (rawConfig: CodeAnalyzerConfig, ruleSelection: RuleSelection) => ConfigModel;
 
-export class DummyConfigModel {
+export class AnnotatedConfigModel implements ConfigModel {
+	private readonly rawConfig: CodeAnalyzerConfig;
+	private readonly ruleSelection: RuleSelection;
+
+	private constructor(rawConfig: CodeAnalyzerConfig, ruleSelection: RuleSelection) {
+		this.rawConfig = rawConfig;
+		this.ruleSelection = ruleSelection;
+	}
+
 	toFormattedOutput(_format: OutputFormat): string {
-		return DUMMY_CONFIG;
+		return yamlFormatConfig(this.rawConfig, this.ruleSelection);
```

Comment:
If we ever end up supporting multiple formats, I'll probably want to move this callstack into a dedicated formatter class, but it's fine for now.